neuralmonkey/dataset.py

```diff
@@ -516,14 +524,35 @@ def itergen():
 
         # Iterate over the rest of the data until buffer is empty
         batch_index = 0
+        buckets = {} \
+            # type: Dict[int, List[Dict[str, Callable[[], Iterator]]]]
```

Comment:
tohle se dělá tak, že si ten typ pojmenuješ v preambuli modulu a tady už jen použiješ to krátký jméno. Preferably to ještě dekomponuješ a pojmenuješ i ty zajímavý typy, ze kterých je to složený.

Dál nechápu, proč je to Dict přes integery. Tomu se ve slušných domácnostech říká List.

Reply:
buckets muze mit max. length 'batch_size // bucket_span'. Z toho vetsinu elementu z toho pole ani nemusis vyuzivat. Prislo mi, ze dict je v tomhle pripade lepsi, i kdyz v nem mas integerove klice.

Reply:
Jestli to je seznam, kde každá položka až do nějakýho maximálního indexu dává smysl, tak to má bejt List.

Reply:
Plus jsi odignoroval možnost ty typy nějak pojmenovat.